Project/PanelInspector.cpp

```diff
@@ -0,0 +1,231 @@
+void PanelInspector::Update()
+{
+	ImGui::SetNextWindowDockID(App->editor->dock_right_id, ImGuiCond_FirstUseEver);
+	if (ImGui::Begin(name, &enabled))
+	{
+		GameObject* selected = App->editor->panel_hierarchy.selected_object;
+		if (selected == nullptr) 
+		{
+			ImGui::End();
+			return;
+		}
+		char name[100];
+		sprintf_s(name, 100, "%s", selected->name.c_str());
+		if (ImGui::InputText("Name", name, 100))
+		{
+			selected->name = name;
+		}
+		bool enable = true;
+		if (ImGui::Checkbox("Enable", &enable))
+		{
+			// TODO: Add enable option to game Objects
+		}
+
+		ImGui::Separator();
+		ComponentTransform* transform = selected->GetComponent<ComponentTransform>();
+		float3 pos3 = transform->GetPosition();
+		float pos[] = { pos3[0], pos3[1], pos3[2] };
+		float3 scale3 = transform->GetScale();
+		float scale[] = { scale3[0], scale3[1], scale3[2] };
+		float3 rotation3 = transform->GetRotation().ToEulerXYZ() * RADTODEG;
+		float rotation[] = { rotation3[0], rotation3[1], rotation3[2] };
+
+		// Clamp Values
+		if (rotation[1] >= 180) rotation[1] = rotation[1] - 360;
+		if (rotation[1] <= -180) rotation[1] = rotation[1] + 360;
+		if (rotation[0] == -0.0f || rotation[0] == -180.0f || rotation[0] == 180.0f) rotation[0] = 0.0f;
+		if (rotation[2] == -0.0f || rotation[2] == -180.0f || rotation[2] == 180.0f) rotation[2] = 0.0f;
+
+		if (ImGui::CollapsingHeader("Transformation")) 
+		{
+			ImGui::TextColored(color, "Transformation (X,Y,Z)");
+			if (ImGui::DragFloat3("Position", pos, drag_speed2f, -inf, inf))
+			{
+				transform->SetPosition(float3(pos[0], pos[1], pos[2]));
+			}
+			if (ImGui::DragFloat3("Scale", scale, drag_speed2f, 0, inf))
+			{
+				transform->SetScale(float3(scale[0], scale[1], scale[2]));
+			}
+			if (ImGui::DragFloat3("Rotation", rotation, drag_speed2f, -inf, inf))
+			{
+				transform->SetRotation(Quat::FromEulerXYZ(rotation[0] * DEGTORAD, rotation[1] * DEGTORAD, rotation[2] * DEGTORAD));
+			}
+		}
+
+		std::vector<ComponentMesh*> meshes = selected->GetComponents<ComponentMesh>();
+		for (std::vector<ComponentMesh*>::iterator it = meshes.begin(); it != meshes.end(); it++)
+		{
+			int count = 1;
+			char name[50];
+			sprintf_s(name, 50, "Mesh Information %d", count);
+			if (*it != nullptr)
+			{
+				// Show only # when multiple
+				if (meshes.size() == 1)
+				{
+					sprintf_s(name, 50, "Mesh Information");
+				}
+				if (ImGui::CollapsingHeader(name))
+				{
+					ImGui::TextColored(color, "Geometry");
+					ImGui::TextWrapped("Num Vertices: ");
+					ImGui::SameLine();
+					ImGui::TextColored(color, "%d", (*it)->num_vertices);
+					ImGui::TextWrapped("Num Triangles: ");
+					ImGui::SameLine();
+					ImGui::TextColored(color, "%d", (*it)->num_indices / 3);
+					ImGui::Separator();
+				}
+				count++;
+			}
+		}
+		std::vector < ComponentMaterial*> materials = selected->GetComponents<ComponentMaterial>();
+		for (std::vector<ComponentMaterial*>::iterator it = materials.begin(); it != materials.end(); it++)
```

Comment:
Usa for-each en lugar de iteradores siempre que sea posible, es mucho más legible:

for (ComponentMaterial* material : materials)